tests/real/community.rs

```diff
@@ -795,46 +800,183 @@ async fn test_jobs_with_blocking_handlers() {
+    )
+    .await
+    .unwrap();
+    c.queue_remove(&[local]).await.unwrap();
 
     let mut w = Worker::builder::<io::Error>()
-        .register_blocking_fn("panic_SYNC_handler", |_j| {
-            panic!("Panic inside sync the handler...");
+        //  sync handlers
+        .register_blocking_fn("panic_SYNC_handler_str", |_j| {
+            panic!("panic_SYNC_handler_str");
+        })
+        .register_blocking_fn("panic_SYNC_handler_String", |_j| {
+            panic_any("panic_SYNC_handler_String".to_string());
+        })
+        .register_blocking_fn("panic_SYNC_handler_int", |_j| {
+            panic_any(0);
+        })
+        .register_blocking_fn("error_from_SYNC_handler", |_j| {
+            Err::<(), io::Error>(io::Error::new(
+                io::ErrorKind::Other,
+                "error_from_SYNC_handler",
+            ))
+        })
+        // async handlers
+        .register_fn("panic_ASYNC_handler_str", |_j| async move {
+            panic!("panic_ASYNC_handler_str");
+        })
+        .register_fn("panic_ASYNC_handler_String", |_j| async move {
+            panic_any("panic_ASYNC_handler_String".to_string());
+        })
+        .register_fn("panic_ASYNC_handler_int", |_j| async move {
+            panic_any(0);
         })
-        .register_fn("panic_ASYNC_handler", |_j| async move {
-            panic!("Panic inside async handler...");
+        .register_fn("error_from_ASYNC_handler", |_j| async move {
+            Err::<(), io::Error>(io::Error::new(
+                io::ErrorKind::Other,
+                "error_from_ASYNC_handler",
+            ))
         })
         .connect()
         .await
         .unwrap();
 
-    let mut c = Client::connect().await.unwrap();
-
-    c.enqueue(Job::builder("panic_SYNC_handler").queue(local).build())
-        .await
-        .unwrap();
+    // let's enqueue jobs first time and ...
+    c.enqueue_many(
+        job_kind_vs_error_msg
+            .keys()
+            .map(|&jkind| Job::builder(jkind).queue(local).args([local]).build())
+            .collect::<Vec<_>>(),
+    )
+    .await
+    .unwrap();
+
+    // ... consume all the jobs from the queue and _fail_ them
+    // "in different ways" (see our worker setup above)
+    for _ in 0..njobs {
+        assert!(w.run_one(0, &[local]).await.unwrap());
+    }
 
-    // we _did_ consume and process the job, the processing result itself though
-    // was a failure; however, a panic in the handler was "intercepted" and communicated
-    // to the Faktory server via the FAIL command;
-    // note how the test run is not interrupted with a panic
-    assert!(w.run_one(0, &[local]).await.unwrap());
+    // let's now make sure all the jobs are re-enqueued
+    c.requeue(
+        MutationTarget::Retries,
+        MutationFilter::builder().pattern(pattern.as_str()).build(),
+    )
+    .await
+    .unwrap();
+
+    // now, let's create a worker who will only send jobs to the
+    // test's main thread to make some assertions
```

Comment:
I'd like a description here about why we have to do this in two passes. I _think_ it's because the failure information in the job isn't something that's passed back to the client, it's something that's passed "forward" to the next handler the job is passed to? But that's not spelled out here.

Reply:
Exactly, and it is also available via Web UI. Add t a comment:
```
we _did_ consume and process the job, the processing result itself though
was a failure; however, a panic in the handler was "intercepted" and communicated
to the Faktory server via the FAIL command, the error message and the backtrace, if any,
will then be available to a Web UI user or to the worker consumes the retried job
(if `Job::retry` is Some and > 0 and there are `Failure::retry_remaining` attempts left)
in this job's `failure` field; see how we are consuming the retried jobs later
in this test and examin the failure details;
```